### PR TITLE
[CDAP-15787] Updated JavaScriptTransform for new validation API

### PR DESCRIPTION
**Pull request by dli357**
Updated the JavaScriptTransform.java class for CDAP-15787

### CODE REVIEW

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (CuriousVini) [inline core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java:149]: Not your change, but change schema == null check to Strings.isNullOrEmpty(schema). We should be checking for both null and empty.

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (CuriousVini) [inline core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: keep corrective action null in this case.

Comment (CuriousVini) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: period at the end of error message

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (CuriousVini) [inline core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: constant

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (CuriousVini) [inline core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: Invalid schema : %s

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (CuriousVini) [inline core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: we should not be changing transform method. I think the changes in init method should be sufficient.

**Review by CuriousVini — COMMENTED** [review]
is existing test passing?

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java:149]: Would we be able to do this? schema is an object and not a string.

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: Fixed

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: Fixed

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: Fixed

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: Fixed

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (CuriousVini) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java:149]: right, this only applies to schema string.

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: nit: fix indentation:

```
emitter.emitError(
  getErrorObject(invalidEntry,
                 decodeRecord((Map) invalidEntry.get("InvalidRecord"), errSchema)));
```

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: nit: indentation:

```
.addFailure("Invalid lookup config."
            "Expected JSON map of string to string.")
  .withConfigProperty(LOOKUP);
```

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: indentation

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: indentation

**Review by albertshau — APPROVED** [review]
lgtm after indentation fixes. Please squash commits before merging.

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: Fixed

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: Fixed

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: Fixed

File: core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java
Comment (dli357) [reply core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java]: Fixed

**Pull request merged**
